src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -324,6 +341,14 @@ public async Task<Document> ApplyCodeFixesForSpecificDiagnosticIdAsync(Document
             }
         }
 
+        private string GetOrComputeCodeFixProviderTypeName(CodeFixProvider provider)
+        {
+            return _fixerTypeNameCache.GetValue(provider, ComputeCodeFixProviderTypeName);
+
+            static string ComputeCodeFixProviderTypeName(CodeFixProvider provider)
+                => provider.GetType().Name;
```

Comment:
this should be super cheap.  not sure why we need to cache this.  i would honeslty expect the cache to be slower here.